### ISSUE TITLE
Invalid gzip header (Gzip::Error)

### ISSUE THREAD

**Issue by jeduardo824** [spoved/vcr.cr#2]
Hi,

I need to use VCR to test an application, but when I run the test suite, after the cassette recording, Gzip throws this unhandled exception:

![Screen Shot 2019-08-20 at 09 35 24](https://user-images.githubusercontent.com/27960597/63347870-73c96080-c32e-11e9-923a-2fb2f5b7a584.png)

Can someone help me?

**Comment by kalinon** [comment]
@jeduardo824 sorry i missed this. I have encountered this error as well. Let me look into it. I solved it by removing the GZip headers on the responses in the fixture files. Dont remember if i pushed or made a release with the fix.

**Comment by kalinon** [comment]
@jeduardo824 I was not able to reproduce with crystal 0.30.1 and halite 0.10.3. As you can see from the fixture i have here, it does have a `Content-Encoding: gzip` but the spec test still passes.

```
HTTP/1.1 200 OK
Date: Sat, 31 Aug 2019 20:48:35 GMT
Content-Type: application/json; charset=utf-8
Transfer-Encoding: chunked
Connection: close
Set-Cookie: __cfduid=dfb88623fe88b78eba85f307e2b41aace1567284515; expires=Sun, 30-Aug-20 20:48:35 GMT; path=/; domain=.typicode.com; HttpOnly
X-Powered-By: Express
Vary: Origin, Accept-Encoding
Access-Control-Allow-Credentials: true
Cache-Control: public, max-age=14400
Pragma: no-cache
Expires: Sun, 01 Sep 2019 00:48:35 GMT
X-Content-Type-Options: nosniff
Etag: W/"53-hfEnumeNh6YirfjyjaujcOPPT+s"
Via: 1.1 vegur
CF-Cache-Status: HIT
Age: 6157
Expect-CT: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
Server: cloudflare
CF-RAY: 50f1ddbe3810c53c-ORD
Content-Encoding: gzip
Content-Length: 83

{
  "userId": 1,
  "id": 1,
  "title": "delectus aut autem",
  "completed": false
}
```

Can you provide your headers? or an endpoint it is failing on?

**Comment by jeduardo824** [comment]
Hi! Yes, here is my header:

```
HTTP/1.1 200 OK
Content-Type: application/json; charset=UTF-8
Date: Wed, 21 Aug 2019 18:58:42 GMT
Expires: Wed, 21 Aug 2019 18:59:12 GMT
Cache-Control: public, max-age=30
Vary: Accept-Language
Server: mafe
Content-Encoding: gzip
Content-Length: 31566
X-XSS-Protection: 0
X-Frame-Options: SAMEORIGIN
Server-Timing: gfet4t7; dur=253
Alt-Svc: quic=":443"; ma=2592000; v="46,43,39"
Connection: close
```

When I deleted the "Content-Encoding" line, the spec passed. Maybe this could be a problem with Net::HTTP or Gzip on Crystal?

**Comment by kalinon** [comment]
Have you tried to update your crystal to 0.30.1?

**Comment by jeduardo824** [comment]
Yes!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by kalinon** [comment]
~~Hmmm. very interesting. Can you provide your `shard.lock` file? i would like to see the dependency versions to see if i can reproduce.~~

nevermind. i can reproduce.

**Comment by kalinon** [comment]
Fixed in 0.2.2

**Comment by jeduardo824** [comment]
Awesome!
